src/keys/slip10.rs

```diff
@@ -363,21 +390,22 @@ impl<K: hazmat::Derivable> Slip10<K> {
         K::is_key_valid(self.key_bytes())
     }
 
-    fn calc_data(&self, hardened: bool) -> [u8; 33] {
-        if hardened {
-            *self.key_bytes()
-        } else {
-            debug_assert!(K::ALLOW_NON_HARDENED);
-            K::calc_non_hardened_data(self.key_bytes())
-        }
+    fn calc_data<S>(&self, segment: S) -> [u8; 33]
+    where
+        S: Segment,
+        K: hazmat::CalcData<S>,
+    {
+        K::calc_data(self.key_bytes(), segment)
     }
 
-    fn derive_child_key(&self, segment: &Segment) -> Self {
-        debug_assert!(K::ALLOW_NON_HARDENED || segment.is_hardened());
-
+    fn derive_child_key<S>(&self, segment: S) -> Self
+    where
+        S: Segment,
+        K: hazmat::CalcData<S>,
+    {
         let mut data = [0u8; 33 + 4];
-        data[..33].copy_from_slice(&self.calc_data(segment.is_hardened()));
-        data[33..].copy_from_slice(&segment.bs()); // ser32(i)
+        data[..33].copy_from_slice(&self.calc_data(segment));
+        data[33..].copy_from_slice(&segment.ser32()); // ser32(i)
```

Comment:
is this a comment?

Reply:
It is. Long time ago it meant something. Removed.